### PR TITLE
Remove elixir server-path variable and add a way to change the lsp spec

### PR DESCRIPTION
**Pull request by Sasanidas**
So... I just realize that the variable only get expanded at compilation time, so when I changed the variable in the config the value doesn't get updated.

The idea is to just change the value in the user configuration with a good old setf command:
```
(defvar *fer/elixir-lsp-path*
  '("sh" "path-to"))

(setf (lem-lsp-mode/spec:spec-command
       (lem-lsp-mode/spec:get-language-spec 'lem-elixir-mode:elixir-mode))
      *fer/elixir-lsp-path*)

```

### CODE REVIEW

**Comment by Sasanidas** [comment]
If you think this makes sense, I'll update the documentation in the page later :+1:

File: modes/lsp-mode/spec.lisp [hunk 1]
```diff
@@ -55,6 +55,10 @@
   (check-type spec spec)
   (setf (get major-mode 'spec) spec))
 
+(defmethod (setf spec-command) ((command list)
+				(spec lem-lsp-mode/spec::spec))
```
Comment (cxxxr) [inline modes/lsp-mode/spec.lisp:59]: Internal prefix maybe not needed.

**Review by cxxxr — APPROVED** [review]
I think it is good!

File: modes/lsp-mode/spec.lisp (hunk 1 above)
Comment (Sasanidas) [reply modes/lsp-mode/spec.lisp]: Done :+1:

**Comment by cxxxr** [comment]
Thank you!

**Pull request merged**